src/ui/utils/useCopy.tsx

```diff
@@ -0,0 +1,58 @@
+import { message } from 'antd';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import IconSuccess from 'ui/assets/success.svg';
+
+interface CopyOptions {
+  variant?: 'address';
+  duration?: number;
+  successText?: string;
+}
+
+export const useCopy = () => {
+  const [hasCopied, setHasCopied] = useState(false);
+  const { t } = useTranslation();
+
+  const copyToClipboard = async (text: string, options: CopyOptions = {}) => {
```

Comment:
maybe you can use https://github.com/RabbyHub/Rabby/blob/develop/src/ui/utils/clipboard.tsx#L33 for instead?
demo: https://github.com/RabbyHub/Rabby/blob/develop/src/ui/views/RabbyPoints/component/CodeAndShare.tsx#L48